src/main/java/com/afornalik/ox/ui/UIDrawBoard.java

```diff
@@ -29,11 +33,26 @@ String draw() {
             if (i < 10) sb.append(i).append(" ");
             else sb.append(i);
             for (int k = 0; k < boardDimension.intValue(); k++) {
-                if (k % 2 == 0) sb.append("|");
-                else sb.append("_");
+                if (k % 2 == 0) {
+                    sb.append("|");
+                } else {
+                    BoardField boardField = board.receiveBoardField(index);
+                    index++;
+                    if (boardField != null) {
+                        if (boardField.showStatus() == FieldStatus.X) {
+                            sb.append("X");
+                        } else {
+                            sb.append("O");
+                        }
+                    } else {
+                        sb.append("_");
+                    }
+
+                }
+
```

Comment:
for if for if if if.. I know I have been there, but try to split it into some helper methods, it would be easier to read

Reply:
Ok i extracted 2 method.

Reply:
Thats better:)